src/pages/Home.tsx

```diff
@@ -140,39 +135,36 @@ function Home() {
   }
 
   async function initializeSarcophagus() {
+    const currentTimestamp = Date.now();
+    const resurrectionTime = currentTimestamp + 7 * 24 * 60 * 60 * 1000;
+    const sarcoId = utils.id(embalmerAddress + currentTimestamp.toString());
+    setCurrentSarcoId(sarcoId);
     const args = [
-      sarcophagusName,
+      sarcoId,
+      {
+        name: sarcophagusName,
+        recipient: recipientAddress,
+        resurrectionTime: resurrectionTime,
+        canBeTransferred: canBeTransferred,
+        minShards: minimumNumberShards,
+      },
       archaeologists,
       arweaveArchaeologist,
-      recipientAddress,
-      resurrectionTime,
-      maxResurrectionInterval,
-      canBeTransferred,
-      minimumNumberShards,
-      sarcoId,
     ];
     await initialize({
       args: args,
       toastText: 'Initialize Sarcophagus',
     });
-    const arweareTxId = await uploadArweaveFile(sarcoId, doubleEncryptedFile || Buffer.from(''));
+
+    const arweareTxId = await uploadArweaveFile(doubleEncryptedFile || Buffer.from(''));
```

Comment:
Why the fallback of `Buffer.from('')` here? If `doubleEncryptedFile` is undefined, do we want to run this `uploadArweaveFile` function?